File: Beecrowd/Beginner/1435 - 1985/1961 - Jumping Frog/Python/1961_JumpingFrog.py
height, pipes = map(int, input().split())

pipes_height = list(map(int, input().split()))

is_possible = True

for i in range(pipes - 1):
    if abs(pipes_height[i] - pipes_height[i + 1]) > height:
        is_possible = False
        break
    
if is_possible:
    print("YOU WIN")
else:
    print("GAME OVER")

File: Beecrowd/Beginner/1435 - 1985/1961 - Jumping Frog/C/1961_JumpingFrog.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int height, pipes;
    scanf("%d %d", &height, &pipes);

    int previousPipe;
    scanf("%d", &previousPipe);

    int isPossible = 1;
    
    for (int i = 1; i < pipes; i++) {
        int currentPipe;
        scanf("%d", &currentPipe);

        if (abs(currentPipe - previousPipe) > height) {
            isPossible = 0;
            break;
        }

        previousPipe = currentPipe;
    }

    if (isPossible) printf("YOU WIN\n");
    else printf("GAME OVER\n");

    return 0;
}
